### PR TITLE
Update/checkout plan messaging

### PR DESCRIPTION
**Pull request by fditrapani**
#### Changes proposed in this Pull Request

This PR cleans up some messaging in checkout so that it doesn't clutter the interface with too much copy.

**Before**
image

**After**
image

#### Testing instructions

* Upgrade a paid plan and confirm the "Discount for first year" messaging shows.
* Upgrade a paid plan purchased with a two year term and confirm "Discount for first term" appears. 
* Add a plan to a free site and confirm this messaging doesn't appear.

### CODE REVIEW

**Review by sirbrillig — APPROVED** [review]
❤️  Looks great.

**Review by michaeldcain — APPROVED** [review]
Just a minor change. Otherwise, LGTM

File: client/my-sites/checkout/composite-checkout/components/wp-order-review-line-items.js [hunk 1]
```diff
@@ -105,8 +105,6 @@ function WPLineItem( {
 						<LineItemSublabelAndPrice item={ item } isMonthlyPricingTest={ isMonthlyPricingTest } />
 						<DomainDiscountCallout item={ item } />
 						{ isMonthlyPricingTest && <AnnualDiscountCallout item={ item } /> }
-					</LineItemMeta>
-					<LineItemMeta>
 						<DiscountForFirstYearOnly item={ item } />
 					</LineItemMeta>
 				</>
```
[the remark this answers is not in the document]
Comment (fditrapani) [reply client/my-sites/checkout/composite-checkout/components/wp-order-review-line-items.js:110]: Nice catch! I've taken it out.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5158321

Thank you @fditrapani for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5158321).

**Pull request merged**